File: TareaGo/Kata04/Clima.go
package main

import (
	"fmt"
	"os"
	"strconv"
)

type Clima struct {
	Dy int
	MxT int
	MnT int
}

func leerArchivo() string{
	contenido, _ := os.Open("weather.dat")
	contenido.Seek(91, 0)

	contenido_bytes := make([]byte, 2699)
	contenido.Read(contenido_bytes)
	contenido.Close()

	cadena := string(contenido_bytes)
	return cadena
}

func convertirArreglo(clima []Clima) []Clima {

	cadena := leerArchivo()
	var cad1, cad2, cad3 string
	
	idx := 0
	var delimitador int
	for x := 0; x < len(clima); x++ {
		delimitador += 4
		for idx < delimitador {
			if cadena[idx] != ' ' {
				cad1 += string(cadena[idx])
			}
			idx++
		}
		delimitador += 4

		for idx < delimitador {
			if cadena[idx] != ' ' {
				cad2 += string(cadena[idx])
			}
			idx++
		}
		delimitador += 6

		for idx < delimitador {
			if cadena[idx] != ' ' {
				if cadena[idx] != '*' {
					cad3 += string(cadena[idx])
				}
			}
			idx++
		}

		delimitador += 76; idx += 76

		clima[x].Dy, _ = strconv.Atoi(cad1)
		clima[x].MxT, _ = strconv.Atoi(cad2)
		clima[x].MnT, _ = strconv.Atoi(cad3)
		cad1 = ""; cad2 = ""; cad3 = ""
	 }

	return clima
}

func imprimirArreglo(clima []Clima) {
	fmt.Println("Dia\tMxT\tMnT")
	for i := 0; i < len(clima); i++ {
		fmt.Printf("%d \t%d \t%d \n", clima[i].Dy, clima[i].MxT, clima[i].MnT)
	}
}

func obtenerMnT(clima []Clima) {
	
	var temp Clima
	temp.MnT = 100

	for i := 0; i < len(clima); i++ {
		if clima[i].MnT < temp.MnT {
			temp = clima[i]
		}
	}

	fmt.Println("Dia: ", temp.Dy, "MnT: ", temp.MnT)
}

func obtenerMxT(clima []Clima) {
	
	var temp Clima
	temp.MxT = 0

	for i := 0; i < len(clima); i++ {
		if clima[i].MxT > temp.MxT {
			temp = clima[i]
		}
	}

	fmt.Println("Dia: ", temp.Dy, "MxT: ", temp.MxT)
}

func main() {

	clima := make([]Clima, 30)

	clima = convertirArreglo(clima)
	imprimirArreglo(clima)
	obtenerMnT(clima)
	obtenerMxT(clima)
}

File: TareaPython/Clima.py
class Temperatura(object):

	def __init__(self, dia, MxT, MnT):
		self.dia = dia
		self.MxT = MxT
		self.MnT = MnT

	def imprimir(self):
		print self.dia, "\t", self.MxT, "\t", self.MnT

class ArregloTemperatura(object):
	"""docstring for ArregloTemperatura"""
	def __init__(self):
		self.arreglo = []

	def leerArchivo(self):
		archivo = open("weather.dat")
		archivo.seek(93, 0)

		for x in range(30):
			dia = int(archivo.read(2))
			archivo.seek(2, 1)
			Mxt = int(archivo.read(2))
			archivo.seek(4, 1)
			Mnt = int(archivo.read(2))
			archivo.seek(78, 1)
			temperatura = Temperatura(dia, Mxt, Mnt)
			self.agregar(temperatura)
		archivo.close()

	def agregar(self, temperatura):
		self.arreglo.append(temperatura)

	def imprimir(self):
		print "Dy\t", "MxT\t", "MnT"
		print("-------------------")
		for x in range(len(self.arreglo)):
			self.arreglo[x].imprimir()

	def obtenerMnT(self):
		temp = Temperatura(0,0,100)

		for x in range(len(self.arreglo)):
			if self.arreglo[x].MnT < temp.MnT:
				temp = self.arreglo[x]
		print "Dia: ", temp.dia, " MnT: ", temp.MnT


	def obtenerMxT(self):
		temp = Temperatura(0,0,0)

		for x in range(len(self.arreglo)):
			if self.arreglo[x].MxT > temp.MxT:
				temp = self.arreglo[x]
		print "Dia: ", temp.dia, " MxT: ", temp.MxT

def main():

	arregloTemperatura = ArregloTemperatura()

	arregloTemperatura.leerArchivo()
	arregloTemperatura.imprimir()
	arregloTemperatura.obtenerMnT()
	arregloTemperatura.obtenerMxT()


if __name__ == '__main__':
	main()
